### ISSUE TITLE
[BUG] Issue with Azure Disks Persistent Volume Attachment in AKS

### ISSUE THREAD

**Issue by dev-nimbeo** [Azure/AKS#4084]
**Description of the bug**
During the deployment of a StatefulSet in Azure Kubernetes Service (AKS), I encountered an issue when using PersistentVolumeClaims (PVCs) with volumeClaimTemplates to provision Azure disks. The pods and the persistent volumes are being deployed correctly in different zones with tier Standard SSD LRS, as storage class was "default".  Specifically, one of the volumes fails to attach to the pod in a specific zone. The error message received is as follows:

> ... failed with Retriable: false, RetryAfter: 0s, HTTPStatusCode: 400, RawError: {
  "error": {
    "code": "InvalidParameter",
    "message": "Storage Account type PremiumV2_LRS is not supported with caching type ReadOnly. Please use caching type None.",
    "target": "dataDisk.caching"
  }
}

It's important to note that the storage account deployed is not premium, which makes the error message confusing.

**Expected behavior**
All volumes should attach to their respective pods without any errors, regardless of their allocation zones, as they are provisioned with Azure disks of the same tier and configuration.

**Environment:**
 - Kubernetes version: 1.25.15

**Comment by andyzhangx** [comment]
@dev-nimbeo 
have you specified `PremiumV2_LRS` in parameters of specific storage class? pls provide the full disk URL, otherwise it's difficult to investigate.

the disk driver would provision standardssd_lrs by default, so  `PremiumV2_LRS` would be probably specified in your setting somewhere.

**Comment by dev-nimbeo** [comment]
No, PremiumV2_LRS is not specified anywhere, that it is the strange thing of the problem. In 2 out of 3 zones it works good. In fact, all the volumes are Standard SSD LRS when I checked so there is no bad deployement.

The only different thing is that in that zone I have another premium v2 LRS storage deployed, but it is not the PV or PVC referenced in the new deployment, it has nothing to do with it. It looks a bug for me.

**Comment by AllenWen-at-Azure** [comment]
@dev-nimbeo, we need more information for further investigation/live troubleshooting. please follow above message from the bot to file a support case to engage AKS support engineer for help. Thanks.
